src/backend/web/routes/posts.js

```diff
@@ -33,7 +33,14 @@ posts.get('/', async (req, res) => {
 });
 
 posts.get('/count', async (req, res) => {
-  res.json(count.getPostsCount());
+  try {
+    await res.json(getPostsCount());
```

Comment:
Are you testing your code locally?  This won't work.  You need to `await` the portion of this that is asynchronous (i.e., returns a Promise).  The call to `res.json(...)` is synchronous, and will block while it runs.  However, the call to `getPostsCount()` is promise-based, and needs to go and talk to the database without blocking the main thread.  As such, you need to do something like:

```js
try {
  res.json(await getPostsCount());
```

You could also break it up a bit if you want:

```js
try {
  const count = await getPostsCount();
  res.json(count);
```

Reply:
Ok thanks will make the changes right now, yes I am running the code locally (testing using npm test) but it does not show any errors with posts.js.

Reply:
Nothing in our tests is calling this route.  You should add a test for this too, which will make it a lot easier to understand if it's working or not.

In our posts test file, you want something like:

```js
  it('returns correct number of posts', async () => {
    const count = await request(app).get('/posts/count');

    expect(res.status).toEqual(200);
    expect(res.get('Content-type')).toContain('application/json');
    expect(res.body).toBe(count);
  });
```

(untested code, fix as makes sense...)

Reply:
Ok I am going to add that now thanks David.